src/client/dfs/dfs.c

```diff
@@ -1246,8 +1246,14 @@ open_file(dfs_t *dfs, dfs_obj_t *parent, int flags, daos_oclass_id_t cid,
 		rc = insert_entry(dfs->layout_v, parent->oh, DAOS_TX_NONE, file->name, len,
 				  DAOS_COND_DKEY_INSERT, entry);
 		if (rc == EEXIST && !oexcl) {
+			int rc2;
+
 			/** just try fetching entry to open the file */
-			daos_array_close(file->oh, NULL);
+			rc2 = daos_array_close(file->oh, NULL);
+			if (rc2 == -DER_NOMEM)
+				rc2 = daos_array_close(file->oh, NULL);
```

Comment:
Just curious calling this twice for ENOME is to fix CI memory leak issues if fault injection happened inside close?

Reply:
yes, test that covers this and found the array bug is at https://github.com/daos-stack/daos/pull/12502

Reply:
correct